addons/hr/models/res_users.py

```diff
@@ -137,7 +137,9 @@ class User(models.Model):
 
     @api.depends_context('uid')
     def _compute_is_system(self):
-        self.write({'is_system': self.env.user._is_system()})
+        is_system = self.env.user._is_system()
+        for user in self:
+            user.is_system = is_system
```

Comment:
@JKE-be why it is not:
```
def _compute_is_system(self):
        for user in self:
            user.is_system = user._is_system()
```

Reply:
If you blame this function, you will see that it is used to know if the current user connected is system or not; and not the user on which we compute this field.
For the same reason that we use the decorator `@api.depends_context('uid')`

It allow to mimic the groups='base.group_system' behavior for a view rendered in sudo.